### PR TITLE
refactor/log,core_tests: adjust logging, impl Debug for Node and Stat…

### PR DESCRIPTION
**Pull request by dhardy**
…eMachine

Test rename is to allow selection via `cargo test ... PATTERN`, which before this would always match at least two other tests.

### CODE REVIEW

**Comment by maidsafe-highfive** [comment]
r? @afck

(maidsafe_highfive has picked a reviewer for you, use r? to override)

File: src/state_machine.rs [hunk 1]
```diff
@@ -111,8 +112,8 @@ impl State {
         }
     }
 
-    fn into_bootstrapped(self, proxy_peer_id: PeerId, proxy_public_id: PublicId) -> Self {
-        match self {
+    fn transition_to_final(self, proxy_peer_id: PeerId, proxy_public_id: PublicId) -> Self {
```
Comment (afck) [inline src/state_machine.rs]: I think `into` communicates better that this consumes `self`.
I also like `bootstrapped` better than `final`, as only `Terminated` is really final.

File: src/state_machine.rs [hunk 2]
```diff
@@ -123,6 +124,19 @@ impl State {
                 }
             }
             _ => unreachable!(),
+        };
+        info!("Transitioned to: {:?}", &new_self);
```
Comment (afck) [inline src/state_machine.rs]: Not sure whether we want to have that user-visible in the vault binaries by default. Maybe it should be `trace!`.

File: src/state_machine.rs (hunk 1 above)
Comment (dhardy) [inline src/state_machine.rs]: I found the name `into_bootstrapped` confusing. If anything, it's converting _from_ the bootstrapped state.

Comment (afck) [inline src/state_machine.rs]: No, it's converting _from_ the `Bootstrapping` state into one of the states that implement the `Bootstrapped` trait (or to `Terminated`.

Comment (Fraser999) [inline src/state_machine.rs]: I guess I agree with @afck on both points really.  Certainly `into` seems to be a standard Rust convention.

**Comment by dhardy** [comment]
Travis shows a failure. But the test passes on my machine on the commit I pushed (c5db457829b28d467d510d814de32d00d4cfc8fe) so I think this is caused by some other ancestor or interaction in 313aa2295c3eae3defccbbd23cbe730b44c3e252.

**Pull request merged**